lib/dry/validation/values.rb

```diff
@@ -51,6 +51,18 @@ def [](*args)
         end
       end
 
+      # @api public
+      def key?(key)
+        return data.key?(key) if key.is_a?(Symbol)
+
+        Schema::Path[key].reduce(data) do |a, e|
+          return false unless a.key?(e)
```

Comment:
this will break if `a` is an array

Reply:
@flash-gordon hah I actually just fixed that